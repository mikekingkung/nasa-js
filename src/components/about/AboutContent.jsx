import React from React;

const AboutContent = () => {
    return (
        <>
            About Page
        </>
    )

export default AboutContent;

}