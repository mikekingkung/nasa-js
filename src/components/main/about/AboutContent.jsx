import React from 'react';
import aboutcontentimage from '../../../images/aboutcontentimage.png';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { content } from '../../../data/content.json';

const AboutContent = () => {

    const useStyles = makeStyles((theme) => ({
        container: {
            height: "100%",
            width: '100%',
            padding: "10px"
           },
         missioncontextcontainer: {
           display: "flex",
           flexDirection: "row",
           justifyContent: "space-between",
           height: "100%",
           width: "100%",
           flex: 1,
           backgroundColor: "black",
           color: "red",
           padding: "20",
           minwidth: "200"
          }
      }));

      const className = useStyles();

    return (
        <div>
        <Grid container className="classes.container">
          <Grid item xs={12} className="classes.missioncontextcontainer">               
             <p>{content.about_copyright}</p>
             <p>{content.about_discover}</p>
             <p>{content.about_topics}</p>
             <p>{content.about_links}</p>
             <p>{content.about_secured}</p>
             <p>{content.about_user_details}</p>

          </Grid>
        </Grid>
        <img
            className="img-fluid"
            src= {aboutcontentimage}
            alt="Learn Space Ltd"
        />
        </div>
    );
}

export default AboutContent;