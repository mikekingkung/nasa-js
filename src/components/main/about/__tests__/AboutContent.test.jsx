import React from 'react';
import { render} from '@testing-library/react';
import AboutContent from '../AboutContent';

describe('<AboutContent Page component', () => {
    it('Correctly renders' ,() => {
        const wrapper = render(<AboutContent/>);
        expect(wrapper).toMatchSnapshot();
    });
});