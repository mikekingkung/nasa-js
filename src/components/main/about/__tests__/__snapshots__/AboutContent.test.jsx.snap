// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutContent Page component Correctly renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiGrid-root classes.container MuiGrid-container"
        >
          <div
            class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p>
              This application makes use of the NASA Open API for images. Content is allowed to be displayed under Nasa Copyright. Users are signed in as Guest and can update the guest log in details
            </p>
            <p>
              The discover page, has a default search option when it loads. Users can update the search term and search NASA image and video archives. At the moment for performance the first 10 hits are displayed. It is possible to 'save details'. The url and title will then be available on the user details page for future reference / export.
            </p>
            <p>
              The topics page provides various links by subject area of interest to NASA resources on the NASA home page.
            </p>
            <p>
              The links page provides 4 different cards of interest. This will be expanded on later in the next phase of the project.
            </p>
            <p>
              The secured page is only visible if keycloak is running and the user has logged in with the correct credentials. The username/password and authentication have yet to be intergrated, hence we we just make use of a default guest log in. This will be integrated in a future phase.
            </p>
            <p>
              The user details page, displays the current Guest user details, along with any links that that have been selected from the Discover page during searching.
            </p>
          </div>
        </div>
        <img
          alt="Learn Space Ltd"
          class="img-fluid"
          src="aboutcontentimage.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiGrid-root classes.container MuiGrid-container"
      >
        <div
          class="MuiGrid-root classes.missioncontextcontainer MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p>
            This application makes use of the NASA Open API for images. Content is allowed to be displayed under Nasa Copyright. Users are signed in as Guest and can update the guest log in details
          </p>
          <p>
            The discover page, has a default search option when it loads. Users can update the search term and search NASA image and video archives. At the moment for performance the first 10 hits are displayed. It is possible to 'save details'. The url and title will then be available on the user details page for future reference / export.
          </p>
          <p>
            The topics page provides various links by subject area of interest to NASA resources on the NASA home page.
          </p>
          <p>
            The links page provides 4 different cards of interest. This will be expanded on later in the next phase of the project.
          </p>
          <p>
            The secured page is only visible if keycloak is running and the user has logged in with the correct credentials. The username/password and authentication have yet to be intergrated, hence we we just make use of a default guest log in. This will be integrated in a future phase.
          </p>
          <p>
            The user details page, displays the current Guest user details, along with any links that that have been selected from the Discover page during searching.
          </p>
        </div>
      </div>
      <img
        alt="Learn Space Ltd"
        class="img-fluid"
        src="aboutcontentimage.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
